### PR TITLE
Use same state object for state and list

### PR DESCRIPTION
**Pull request by crosbymichael**
Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

**Comment by mikebrow** [comment]
Reviewed. Looks good to me.

**Comment by mrunalp** [comment]
One of the integration tests is failing.

File: tests/integration/list.bats [hunk 1]
```diff
@@ -47,7 +47,7 @@ function teardown() {
...
-  [[ "${lines[0]}" == *[,][\{]"\"id\""[:]"\"test_box2\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"created\""[:]*[0-9]*[\}]* ]]
-  [[ "${lines[0]}" == *[,][\{]"\"id\""[:]"\"test_box3\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"created\""[:]*[0-9]*[\}][\]] ]]
+  [[ "${lines[0]}" == [\[][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box1\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}]* ]]
+  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box2\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}]* ]]
+  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box3\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}][\]] ]]
```
Comment (crosbymichael) [inline tests/integration/list.bats]: @mikebrow any idea why the last line on the `list` test is incorrect?  I cannot figure out why its making this fail

Comment (mikebrow) [inline tests/integration/list.bats]: checking...

Comment (mikebrow) [inline tests/integration/list.bats]: Here you go:
```
diff --git a/tests/integration/list.bats b/tests/integration/list.bats
index bf05a0c..73ca77a 100644
--- a/tests/integration/list.bats
+++ b/tests/integration/list.bats
@@ -47,7 +47,7 @@ function teardown() {
 
   ROOT=$HELLO_BUNDLE runc list --format son
   [ "$status" -eq 0 ]
-  [[ "${lines[0]}" == [\[][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box1\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}]* ]]
-  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box2\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}]* ]]
-  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]*[0-9]*[,]"\"id\""[:]"\"test_box3\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]*[0-9]*[,]"\"created\""[:]*[0-9]*[\}][\]] ]]
+  [[ "${lines[0]}" == [\[][\{]"\"ociVersion\""[:]"\""*[0-9][\.]*[0-9][\.]*[0-9]*"\""[,]"\"id\""[:]"\"test_box1\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]"\""*"\""[,]"\"created\""[:]*[0-9]*[\}]* ]]
+  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]"\""*[0-9][\.]*[0-9][\.]*[0-9]*"\""[,]"\"id\""[:]"\"test_box2\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]"\""*"\""[,]"\"created\""[:]*[0-9]*[\}]* ]]
+  [[ "${lines[0]}" == *[,][\{]"\"ociVersion\""[:]"\""*[0-9][\.]*[0-9][\.]*[0-9]*"\""[,]"\"id\""[:]"\"test_box3\""[,]"\"pid\""[:]*[0-9][,]"\"status\""[:]*"\"running\""[,]"\"bundle\""[:]*$BUSYBOX_BUNDLE*[,]"\"rootfs\""[:]"\""*"\""[,]"\"created\""[:]*[0-9]*[\}][\]] ]]
 }

```

**Comment by mrunalp** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**